### PR TITLE
Support multi-part uploads

### PR DESCRIPTION
**Pull request by jbencook**
For large datasets, the current `Session.upload_data` method fails. This PR switches the call to `Object.upload_file` which can do multi-part uploads. Also updated the unit tests.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -20,3 +20,4 @@ examples/tensorflow/distributed_mnist/data
 doc/_build
 **/.DS_Store
 venv/
+*~
```
Comment (owen-t) [inline .gitignore]: Can this make it into the commit message as well.

**Comment by owen-t** [comment]
Thanks for your submission!

File: .gitignore (hunk 1 above)
Comment (jbencook) [reply .gitignore]: Yeah I accidentally added an emacs file on our fork. You mean you want a new commit in this PR with a message about that change?

Comment (owen-t) [reply .gitignore]: It's a good addition, happy for it to be part of this PR. If we leave it in this PR, can the commit message just reference the addition. Something like

"Upload files to S3 using multipart uploads.

Emacs temporary files covered in .gitignore"

Comment (jbencook) [reply .gitignore]: Yeah I'm happy to add it. Any idea if amending my commit message and force pushing to our fork will do it?

Comment (owen-t) [reply .gitignore]: Yeah, should be fine

Comment (jbencook) [reply .gitignore]: Ok that worked

**Pull request merged**